src/test/kotlin/com/terraformation/backend/db/DatabaseBackedTest.kt

```diff
@@ -3047,6 +3051,128 @@ abstract class DatabaseBackedTest {
     return row.id!!.also { inserted.variableWorkflowHistoryIds.add(it) }
   }
 
+  /**
+   * Asserts that a table (or a filtered subset of it) contains an expected set of records and
+   * nothing else.
+   *
+   * @param table The table whose contents should be examined.
+   * @param expected The set of records the table should contain.
+   * @param message Assertion failure message; defaults to the table name.
+   * @param where Optional query condition to assert on a subset of a table's contents. Only rows
+   *   matching the condition will be considered.
+   * @param clearPrimaryKeys If true, the primary key field(s) of the records from the database will
+   *   be cleared before comparing against [expected]. This can be used to ignore database-generated
+   *   IDs.
+   * @param transform Function to apply to records from the database before comparing them to
+   *   [expected]. Can be used to clear or hardwire specific fields that aren't relevant to the
+   *   behavior being tested.
+   */
+  protected fun <R : UpdatableRecord<R>> assertTableEquals(
+      table: Table<R>,
+      expected: Set<R>,
+      message: String = table.name,
+      where: Condition? = null,
+      clearPrimaryKeys: Boolean = false,
```

Comment:
Is this name too verbose? I also considered `nullIds` but that seemed less clear to me (not least because the lower-case "L" looks too similar to the upper-case "I" in some fonts.)

Reply:
perhaps we can use `includePrimaryKeys = true`? I think it is clear enough either way!

Reply:
Good call. "Include primary keys" is a bit more intuitive.